datamade/bga-payroll
# Conditionally show donate banner

## Description

This PR installs [`django-extra-settings`](https://github.com/fabiocaccamo/django-extra-settings) to enable editing of a new setting, `PAYROLL_SHOW_DONATION_BANNER`, via the admin interface. It also uses this setting to conditionally display the large and small donate callouts on the homepage. Finally, it adds a new cache, `vary_on_setting`, stores the homepage there, and extends the Setting admin to clear this special cache when settings are changed, so the banners are toggled in real time.

## Testing instructions

- Deploy to the staging site.
- View the homepage and toggle the year on the "What's inside?" chart so responses for both years are cached.
- Navigate to the admin interface and create the new variable. Set and unset it. Confirm that the donate banner on the homepage is hidden and shown as expected.
- Toggle the year for the "What's inside?" chart and confirm that it responds quickly, i.e., the default cache is _not_ cleared by changing settings.
- Visit the flush route and confirm everything is reloaded from scratch.
